HELP.md

```diff
@@ -0,0 +1,109 @@
+# Contract best practices
+
+## Init once
+
+Make sure you check that state doesn't exists in `init` using
+
+```rust
+assert!(!env::state_exists(), "The contract is already initialized");
+```
+
+
+## Prevent Default
+
+By default `near_sdk` allows contract to be initialized with default state. Usually, if you have a constructor, you want to
+prevent default state initialization.
+
+```rust
+impl Default for Contract {
+    fn default() -> Self {
+        env::panic(b"The contract should be initialized before usage")
+    }
+}
+```
+
+
+## Public vs private methods
+
+`pub fn` makes a method public and exposes it in a contract. It means anyone can call it.
+`fn` makes the method private and it's not exposed from the contract. No one can call it directly. It can only be called
+within a contract directly (not through a promise).
+
+
+## Private callbacks
+
+If you're using callbacks, makes sure you check the caller.
+
+```rust
+assert_eq!(env::current_account_id(), env::predecessor_account_id(), "Can only be called within a contract");
+```
+
+Callbacks have to be public methods.
+
+
+## JSON types
+
+NEAR currently expects contracts to support JSON serialization. JSON can't handle large integers (above 2**53 bits).
+That's why you should use helper classes from the `json_types` in `near_sdk` for `u64` and `u128`.
+We provide types `U64` and `U128`, that wraps the integer into a struct and implements JSON serialization and
+deserialization as a base-10 strings.
+
+E.g.
+
+```rust
+pub fn add(&self, a: U128, b: U128) -> U128 {
+    (a.0 + b.0).into()
```

Comment:
I found this to be annoying.  Is there really not a way to handle this with the bindgen?

Reply:
Not easily. We don't want to make it a requirement to serialize long numbers as strings. Hopefully JSON standard can improve in the future and we can use regular long numbers with string quotes.

Reply:
We can implement math on `U128`, so you'll be able to do simply `a + b`

Reply:
Yeah that's a good idea.